### DISCUSSION TITLE
Multiple Shipping Options - but none viewed on checkout

### DISCUSSION THREAD

**Discussion by inu-web** [openstore-ecommerce/OpenStore#183] [Q&A]
Hi All!
I have configured multiple Shipping options and they all show up under BO/Admin/Plugins and BO/Shipping correctly 
I configured them for different Country codes and shipping rates by price including free ship range settings

![image](https://user-images.githubusercontent.com/7883518/129889533-492321c6-4481-42bf-8ddc-d5f068a62259.png)

![image](https://user-images.githubusercontent.com/7883518/129889874-5aa91d51-0811-4008-8a62-294bc353fe7b.png)

But when I go through the purchasing process and reach the CheckoutSummary.cshtml template, there are no shipping options showing up.

    <div id="shippingmethods">
        <div class="formrow">
            <h3>@ResourceKey("General.shippingmethods")</h3>
            <div id="providerselection">
                @ShippingProviderList(info, "class='shippingmethodselect'")
            </div>
            <div id="shipprovidertemplates"></div>
        </div>
    </div>

The network panel of the browser shows that the request to the handler was successful (200-OK) but the response is empty
![image](https://user-images.githubusercontent.com/7883518/129890309-c7db544c-0638-4fd1-a750-da27b9bfe470.png)

and the area that should show the radio options for shipping variants are empty too:
![image](https://user-images.githubusercontent.com/7883518/129890401-0f70d74b-0795-4c17-83ec-43f68b586a8f.png)

![image](https://user-images.githubusercontent.com/7883518/129890430-b3f89ada-c8f8-4891-9863-db6faac8007f.png)

Can please somebody help or advise what i can check or do to narrow this problem down?
Thanks in advance for any help
kind regards
Gerald

**Comment by inu-web** [comment]
Okay, after a few trial and error tests it turns out that the "valid country codes" do not work as expected.
When I remove everything from this textbox, the shipping method appears in the list
![image](https://user-images.githubusercontent.com/7883518/129896177-9be8849f-9846-43f1-bb2a-447232727f6f.png)
![image](https://user-images.githubusercontent.com/7883518/129896288-299f8369-9b19-43f9-9c69-c26dc95144dc.png)
But as expected then all shipping options appear, because they have no configuration for "valid country codes" where they should show up

Even entering all the other country codes in the textbox for invalid Country codes doesnt change anything - all options still show up

Any idea ?
thanks and regards
Gerald

**Comment by DNNMonster** [comment]
I tried but I am not able to reproduce this which may suggest it's data related.  If you have your dev environment up then you could place a break point in the ShippingProviderLIst function and have a good shot at finding the root cause.  If someone else has seen this then hopefully they will chime in. 
 https://github.com/openstore-ecommerce/OpenStore/blob/a716e0653951f6dbf542224a457abcdc2f51ed40/render/NBrightBuyRazorTokens.cs#L1624

**Reply by inu-web** [reply]
Thank you for your response, but I was not able to find out whats the issue
So I left the configuration for allowed countrycodes blank in BO/ShippingProviders - which then showed me all providers in the checkout.
I solved it with some clientside Code which hides the Shipping providers that are not possible for the selected country
Works for me now ... with this workaround
thank you for taking the time to assist! Your involvement is highly appreciated.
kind regards
Gerald
